Projects/Features/SearchFeature/Sources/ViewControllers/SearchViewController.swift

```diff
@@ -81,55 +73,85 @@ extension SearchViewController {
+        self.view.backgroundColor = .clear
+
         self.add(asChildViewController: beforeVc)
-        beforeVc.delegate = self
     }
 
-    private func bindSubView(_ afterSearch: Bool) {
-        /*
-         A:부모
-         B:자식
+    override public func bind(reactor: SearchReactor) {
+        super.bind(reactor: reactor)
+    }
 
-         서브뷰 추가
-         A.addChild(B)
-         A.view.addSubview(B.view)
-         B.didMove(toParent: A)
+    override public func bindState(reactor: SearchReactor) {
+        super.bindState(reactor: reactor)
 
-         서브뷰 삭제
-         B.willMove(toParent: nil) // 제거되기 직전에 호출
-         B.removeFromParent() // parentVC로 부터 관계 삭제
-         B.view.removeFromSuperview() // parentVC.view.addsubView()와 반대 기능
+        let currentState = reactor.state.share(replay: 1)
 
-         */
+        currentState
+            .map { ($0.typingState, $0.text) }
+            .withUnretained(self)
+            .bind(onNext: { owner, data in
 
-        if let nowChildVc = children.first as? BeforeSearchContentViewController {
-            if afterSearch == false {
-                return
+                let (state, text) = data
 
-            } else {
-                self.remove(asChildViewController: beforeVc)
-                self.add(asChildViewController: afterVc)
-                afterVc.input.text.accept(viewModel.input.textString.value)
-            }
-        } else if let nowChildVc = children.first as? AfterSearchViewController {
-            if afterSearch == true {
-                return
+                owner.cancelButton.alpha = state == .typing ? 1.0 : .zero
+                owner.reactSearchHeader(state)
+                owner.bindSubView(state)
 
-            } else {
-                self.remove(asChildViewController: afterVc)
-                self.add(asChildViewController: beforeVc)
-                beforeVc.delegate = self
-            }
-        }
+                if state == .search {
+                    if text.isEmpty {
+                        guard let textPopupViewController = owner.textPopUpFactory.makeView(
+                            text: "검색어를 입력해주세요.",
+                            cancelButtonIsHidden: true,
+                            allowsDragAndTapToDismiss: nil,
+                            confirmButtonText: nil,
+                            cancelButtonText: nil,
+                            completion: nil,
+                            cancelCompletion: nil
+                        ) as? TextPopupViewController else {
+                            return
+                        }
+                        owner.showPanModal(content: textPopupViewController)
+
+                    } else {
+                        PreferenceManager.shared.addRecentRecords(word: text)
+                        UIView.setAnimationsEnabled(false)
+                        owner.view.endEditing(true)
+                        UIView.setAnimationsEnabled(true)
+                    }
+                }
+
```

Comment:
```suggestion
                guard state == .search else {
                    return
                }

                if text.isEmpty {
                    guard let textPopupViewController = owner.textPopUpFactory.makeView(
                        text: "검색어를 입력해주세요.",
                        cancelButtonIsHidden: true,
                        allowsDragAndTapToDismiss: nil,
                        confirmButtonText: nil,
                        cancelButtonText: nil,
                        completion: nil,
                        cancelCompletion: nil
                    ) as? TextPopupViewController else {
                        return
                    }
                    owner.showPanModal(content: textPopupViewController)
                } else {
                    PreferenceManager.shared.addRecentRecords(word: text)
                    UIView.setAnimationsEnabled(false)
                    owner.view.endEditing(true)
                    UIView.setAnimationsEnabled(true)
                }
```
else 행동이 없으니 guard clause로 조건처리하는거 어떤가요? indent를 더 줄일 수 있어서요

Reply:
적용 완료